File: 2022_convert_1d_array_into_2d_array/dart/solution.dart
class Solution {
  List<List<int>> constuct2DArray(List<int> data, int cols, int rows) {
    List<List<int>> result = [];

    if (data.length != cols * rows) {
      return result;
    }

    for (int i = 0; i < rows * cols; i + rows) {
      result.add(data.sublist(i, i + rows));
    }

    return result;
  }
}

File: 2022_convert_1d_array_into_2d_array/python/solution.py
from typing import List


class Solution:
    def construct2DArray(self, original: List[int], cols: int, rows: int) -> List[List[int]]:
        if rows * cols != len(original):
            return []

        result = []

        start = 0
        end = rows

        while end < rows * cols + 1:
            result.append(original[start:end])
            start = end
            end = end + rows

        return result
